### ISSUE TITLE
Error when installing base_module_record

### ISSUE THREAD

**Issue by bennals** [JayVora-SerpentCS/SerpentCS_Contributions-v8#166]
The following error occurred when installing in a freshly installed Odoo 11 CE (no demo data) on Ubuntu 16.04 using Yenthe666's script [here](https://www.odoo.yenthevg.com/installing-odoo-11-enterprise-ubuntu/).

I read that the only dependency is base so I have not installed any of the other modules in the SerpentCS_Contributions package.

Thanks, 

Bill.

Error:
Odoo Server Error

Traceback (most recent call last):
  File "/odoo/odoo-server/odoo/models.py", line 967, in _validate_fields
    check(self)
  File "/odoo/odoo-server/odoo/addons/base/ir/ir_ui_view.py", line 338, in _check_xml
    self.postprocess_and_fields(view.model, view_doc, view.id)
  File "/odoo/odoo-server/odoo/addons/base/ir/ir_ui_view.py", line 957, in postprocess_and_fields
    self.raise_view_error(_('Model not found: %(model)s') % dict(model=model), view_id)
  File "/odoo/odoo-server/odoo/addons/base/ir/ir_ui_view.py", line 506, in raise_view_error
    raise ValueError(message)
ValueError: Model not found: base.module.record

Error context:
View `base_module_record_objects`
[view_id: 2136, xml_id: n/a, model: base.module.record, parent_id: n/a]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/odoo/odoo-server/odoo/tools/convert.py", line 741, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/odoo/odoo-server/odoo/tools/convert.py", line 651, in _tag_record
    id = self.env(context=rec_context)['ir.model.data']._update(rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode)
  File "/odoo/odoo-server/odoo/addons/base/ir/ir_model.py", line 1469, in _update
    record = record.create(values)
  File "/odoo/odoo-server/odoo/addons/base/ir/ir_ui_view.py", line 401, in create
    return super(View, self).create(self._compute_defaults(values))
  File "/odoo/odoo-server/odoo/models.py", line 3286, in create
    self._fields[key].determine_inverse(record)
  File "/odoo/odoo-server/odoo/fields.py", line 1081, in determine_inverse
    getattr(records, self.inverse)()
  File "/odoo/odoo-server/odoo/addons/base/ir/ir_ui_view.py", line 269, in _inverse_arch
    view.write(data)
  File "/odoo/odoo-server/odoo/addons/base/ir/ir_ui_view.py", line 417, in write
    return super(View, self).write(self._compute_defaults(vals))
  File "/odoo/odoo-server/odoo/models.py", line 3008, in write
    self._write(old_vals)
  File "/odoo/odoo-server/odoo/models.py", line 3161, in _write
    self._validate_fields(vals)
  File "/odoo/odoo-server/odoo/models.py", line 971, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.exceptions.ValidationError: ('Error while validating constraint\n\nModel not found: base.module.record\n\nError context:\nView `base_module_record_objects`\n[view_id: 2136, xml_id: n/a, model: base.module.record, parent_id: n/a]', None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/odoo/odoo-server/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/odoo/odoo-server/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/odoo/odoo-server/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/odoo/odoo-server/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/odoo/odoo-server/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/odoo/odoo-server/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/odoo/odoo-server/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/odoo/odoo-server/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/odoo/odoo-server/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/odoo/odoo-server/addons/web/controllers/main.py", line 934, in call_button
    action = self._call_kw(model, method, args, {})
  File "/odoo/odoo-server/addons/web/controllers/main.py", line 922, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/odoo/odoo-server/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/odoo/odoo-server/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-39>", line 2, in button_immediate_install
  File "/odoo/odoo-server/odoo/addons/base/module/module.py", line 71, in check_and_log
    return method(self, *args, **kwargs)
  File "/odoo/odoo-server/odoo/addons/base/module/module.py", line 448, in button_immediate_install
    return self._button_immediate_function(type(self).button_install)
  File "/odoo/odoo-server/odoo/addons/base/module/module.py", line 541, in _button_immediate_function
    modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "/odoo/odoo-server/odoo/modules/registry.py", line 85, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/odoo/odoo-server/odoo/modules/loading.py", line 343, in load_modules
    loaded_modules, update_module)
  File "/odoo/odoo-server/odoo/modules/loading.py", line 242, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/odoo/odoo-server/odoo/modules/loading.py", line 156, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='data')
  File "/odoo/odoo-server/odoo/modules/loading.py", line 94, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/odoo/odoo-server/odoo/tools/convert.py", line 788, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/odoo/odoo-server/odoo/tools/convert.py", line 849, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/odoo/odoo-server/odoo/tools/convert.py", line 748, in parse
    exc_info[2]
  File "/odoo/odoo-server/odoo/tools/pycompat.py", line 86, in reraise
    raise value.with_traceback(tb)
  File "/odoo/odoo-server/odoo/tools/convert.py", line 741, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/odoo/odoo-server/odoo/tools/convert.py", line 651, in _tag_record
    id = self.env(context=rec_context)['ir.model.data']._update(rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode)
  File "/odoo/odoo-server/odoo/addons/base/ir/ir_model.py", line 1469, in _update
    record = record.create(values)
  File "/odoo/odoo-server/odoo/addons/base/ir/ir_ui_view.py", line 401, in create
    return super(View, self).create(self._compute_defaults(values))
  File "/odoo/odoo-server/odoo/models.py", line 3286, in create
    self._fields[key].determine_inverse(record)
  File "/odoo/odoo-server/odoo/fields.py", line 1081, in determine_inverse
    getattr(records, self.inverse)()
  File "/odoo/odoo-server/odoo/addons/base/ir/ir_ui_view.py", line 269, in _inverse_arch
    view.write(data)
  File "/odoo/odoo-server/odoo/addons/base/ir/ir_ui_view.py", line 417, in write
    return super(View, self).write(self._compute_defaults(vals))
  File "/odoo/odoo-server/odoo/models.py", line 3008, in write
    self._write(old_vals)
  File "/odoo/odoo-server/odoo/models.py", line 3161, in _write
    self._validate_fields(vals)
  File "/odoo/odoo-server/odoo/models.py", line 971, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.tools.convert.ParseError: "Error while validating constraint

Model not found: base.module.record

Error context:
View `base_module_record_objects`
[view_id: 2136, xml_id: n/a, model: base.module.record, parent_id: n/a]
None" while parsing /odoo/odoo-server/addons/base_module_record/wizard/base_module_record_object_view.xml:4, near
<record id="base_module_record_objects_view" model="ir.ui.view">
            <field name="name">base_module_record_objects</field>
            <field name="model">base.module.record</field>
            <field name="arch" type="xml">
                <form string="Objects Recording">
                    <group col="4">
                        <field name="check_date"/>
                        <field name="filter_cond"/>
                    </group>
                    <separator string="Choose objects to record" colspan="4"/>
                    <field name="objects" colspan="4" nolabel="1"/>
                    <group>
                        <field name="info_yaml"/>
                    </group>
                    <footer>
                        <button name="record_objects" string="Record" type="object" class="oe_highlight"/>
                        or
                        <button string="Cancel" class="oe_link" special="cancel"/>
                    </footer>
                </form>
            </field>
        </record>

**Comment by JayVora-SerpentCS** [comment]
The model already exists. I guess you need to restart the server once.

**Comment by JayVora-SerpentCS** [comment]
@arpit983 Can you please check?

**Comment by JayVora-SerpentCS** [comment]
Fixed, check with latest version #172

**Comment by simonitfi** [comment]
You should also apply this 
https://github.com/theopolisme/webcam-streamer/issues/10

**Comment by bennals** [comment]
@JayVora-SerpentCS 
Sorry for the delayed response.  I tried the latest commit and now it installs without errors.  Thank you.  I noticed though that the image links in the module description in the 'Apps' menu are broken.  Don't know if that's important but though it was worth mentioning.
![screenshot](https://user-images.githubusercontent.com/7786180/35803459-a2180b72-0abf-11e8-9145-197245278598.png)

**Comment by JayVora-SerpentCS** [comment]
This must work @bennals , the image exists already.

**Comment by bennals** [comment]
@JayVora-SerpentCS 
Ok, perhaps it's something to do with my server structure.  I have the base_module_record directory in a third party addons folder (/home/odoo/thirdparty/addons) which I have included in the odoo .conf file.

The strange thing is it's the only module that is behaving that way, including other third party modules that I have installed.

![broken_link](https://user-images.githubusercontent.com/7786180/35885637-37430f38-0bda-11e8-9d02-a74661badebd.png)
![no_broken_link](https://user-images.githubusercontent.com/7786180/35885638-37922e4c-0bda-11e8-92aa-682c68874096.png)

Anyway, it's only a cosmetic issue for me, so no need to worry about it really.
